component/azstorage/utils_test.go

```diff
@@ -262,6 +262,29 @@ func (s *utilsTestSuite) TestBfsProxyOptions() {
 	assert.EqualValues(ro.MaxTries, 3)
 	assert.NotEqual(po.RequestLog.SyslogDisabled, true)
 }
+
+type endpointAccountType struct {
+	endpoint string
+	account  AccountType
+	result   string
+}
+
+func (s *utilsTestSuite) TestFormatEndpointAccountType() {
+	assert := assert.New(s.T())
+	var inputs = []endpointAccountType{
+		{endpoint: "https://account.blob.core.windows.net", account: EAccountType.BLOCK(), result: "https://account.blob.core.windows.net"},
+		{endpoint: "https://account.dfs.core.windows.net", account: EAccountType.BLOCK(), result: "https://account.blob.core.windows.net"},
+		{endpoint: "https://account.blob.core.windows.net", account: EAccountType.ADLS(), result: "https://account.dfs.core.windows.net"},
+		{endpoint: "https://account.dfs.core.windows.net", account: EAccountType.ADLS(), result: "https://account.dfs.core.windows.net"},
```

Comment:
Shall we have a test case to mimic our actual code where we first modify based on protocol and then based on account type.

Reply:
We can add a unit test in Azstorage